### PR TITLE
[EDX-461,EDX-490] - Methods and properties in API Ref pages should have Copy Link and appear in the RHS nav

### PR DESCRIPTION
**Pull request by aralovelace**
> _NOTE TO REVIEWERS_ - please **do not review PRs in the `DRAFT` state**, as the PR may change substantially before it is ready to review. Thanks.

## Description

A PR description indicating the purpose of the PR.

* [EDX-490](https://ably.atlassian.net/browse/EDX-490?atlOrigin=eyJpIjoiZmFhNTNlYjFjNGQ1NDc1OWIzYWE5Yzg0NjczMjBmOTciLCJwIjoiaiJ9)
* [EDX-461](https://ably.atlassian.net/browse/EDX-461?atlOrigin=eyJpIjoiZDkyNWNhNGYwY2JkNDgyMzlhM2FjZjVlZjUwYmMwOTQiLCJwIjoiaiJ9)

## Review

Instructions on how to review the PR. 
- https://ably-docs-2-edx-461-hxzn0rvy50.herokuapp.com/docs/api/realtime-sdk#connection
--  should have the new copy link and appear in the RHS
- https://ably-docs-2-edx-461-hxzn0rvy50.herokuapp.com/docs/api/rest-api
-- the H6 should not have Copy Link and not appear in the RHS

### CODE REVIEW

**Review by Srushtika — CHANGES_REQUESTED** [review]
Hey @aralovelace, just noticed that this gets super messy for SSE API and REST API pages as it shows the routes instead of methods and properties. Is it possible to skip those two pages from this logic? cc: @MymmiJ 

- https://ably-docs-next-staging.herokuapp.com/docs/api/rest-api
- https://ably-docs-next-staging.herokuapp.com/docs/api/sse?lang=python

**Comment by aralovelace** [comment]
> Hey @aralovelace, just noticed that this gets super messy for SSE API and REST API pages as it shows the routes instead of methods and properties. Is it possible to skip those two pages from this logic? cc: @MymmiJ
> 
> * https://ably-docs-next-staging.herokuapp.com/docs/api/rest-api
> * https://ably-docs-next-staging.herokuapp.com/docs/api/sse?lang=python

@MymmiJ  hmmm how can we differentiate the use of H6 per page? aside from getting adding a condition to to check if the sitename is not [/api/rest-api](https://ably-docs-next-staging.herokuapp.com/docs/api/rest-api) and [api/sse?lang=python](https://ably-docs-next-staging.herokuapp.com/docs/api/sse?lang=python) ? and I know this would be a messy process

**Comment by aralovelace** [comment]
@Srushtika  @MymmiJ  this is updated now

File: data/createPages/createContentMenuDataFromPage.js [hunk 1]
```diff
@@ -52,23 +64,28 @@ export const createContentMenuDataFromPage = (page, contentMenuData = [], langua
...
+      let newID = id;
+      if (!id) {
+        newID = idFromName(menuItemName);
+        if (page.attribs) {
+          page.attribs.id = newID;
+        } else {
+          page.attribs = {
+            id: newID,
+          };
+        }
```
Comment (snikidev) [inline data/createPages/createContentMenuDataFromPage.js:77]: this seems exactly the same 🤔 or am I missing something?

File: data/createPages/createContentMenuDataFromPage.js [hunk 2]
```diff
@@ -52,23 +64,28 @@ export const createContentMenuDataFromPage = (page, contentMenuData = [], langua
...
-    if (!id) {
-      newID = idFromName(menuItemName);
-      if (page.attribs) {
-        page.attribs.id = newID;
-      } else {
-        page.attribs = {
-          id: newID,
-        };
+    if (isItemBeAddedInNav(menuItemName, name)) {
+      let newID = id;
```
Comment (snikidev) [inline data/createPages/createContentMenuDataFromPage.js:68]: i think we can simplify it to 
```
const newId = id ?? idFromName(menuItemName)
```

File: data/createPages/createContentMenuDataFromPage.js (hunk 1 above)
Comment (snikidev) [reply data/createPages/createContentMenuDataFromPage.js:77]: ahhhh I get it now. Well, we can also do this
```
page.attribs = {
  ...(page.attribs || {}),
  id: idFromName(menuItemName)
}
```

**Review by snikidev — APPROVED** [review]
Love it! 🔥

File: data/createPages/createContentMenuDataFromPage.js [hunk 3]
```diff
@@ -2,10 +2,23 @@ import { isArray } from 'lodash';
...
   [HtmlDataTypes.h2]: 2,
   [HtmlDataTypes.h3]: 3,
+  [HtmlDataTypes.h6]: 3,
+};
+
+const httpRESTfulMethods = ['POST', 'GET', 'PUT', 'PATCH', 'DELETE'];
+
+const isItemBeAddedInNav = (name, header) => {
+  if (header === HtmlDataTypes.h6 && name !== '') {
+    const isRestFUL = httpRESTfulMethods.some((method) => name.toUpperCase().startsWith(method));
```
Comment (MymmiJ) [inline data/createPages/createContentMenuDataFromPage.js:16]: Shouldn't this just be `name`? I don't think we want anything starting with `get` to be treated the same as `GET`, it's not really REST associated in the same reliable way

**Review by MymmiJ — APPROVED** [review]

**Pull request merged**